cmd/cosign/cli/verify/verify_bundle.go

```diff
@@ -36,169 +32,17 @@ import (
-			return newExpiringKey, nil
-		})
-	}
-
-	identityPolicies := []verify.PolicyOption{}
-
-	verificationMaterial := bundle.GetVerificationMaterial()
-
-	if verificationMaterial == nil {
-		return nil, fmt.Errorf("no verification material in bundle")
-	}
-
-	if verificationMaterial.GetPublicKey() != nil {
-		identityPolicies = append(identityPolicies, verify.WithKey())
-	} else {
-		sanMatcher, err := verify.NewSANMatcher(certIdentity, certIdentityRegexp)
-		if err != nil {
-			return nil, err
-		}
-
-		issuerMatcher, err := verify.NewIssuerMatcher(certOIDCIssuer, certOIDCIssuerRegex)
-		if err != nil {
-			return nil, err
-		}
-
-		extensions := certificate.Extensions{
-			GithubWorkflowTrigger:    githubWorkflowTrigger,
-			GithubWorkflowSHA:        githubWorkflowSHA,
-			GithubWorkflowName:       githubWorkflowName,
-			GithubWorkflowRepository: githubWorkflowRepository,
-			GithubWorkflowRef:        githubWorkflowRef,
-		}
-
-		certIdentity, err := verify.NewCertificateIdentity(sanMatcher, issuerMatcher, extensions)
-		if err != nil {
-			return nil, err
-		}
-
-		identityPolicies = append(identityPolicies, verify.WithCertificateIdentity(certIdentity))
-	}
-
-	// Make some educated guesses about verification policy
-	verifierConfig := []verify.VerifierOption{}
-
-	if len(trustedroot.RekorLogs()) > 0 && !ignoreTlog {
-		verifierConfig = append(verifierConfig, verify.WithTransparencyLog(1), verify.WithIntegratedTimestamps(1))
-	}
-
-	if len(trustedroot.TimestampingAuthorities()) > 0 && useSignedTimestamps {
-		verifierConfig = append(verifierConfig, verify.WithSignedTimestamps(1))
-	}
-
-	if !ignoreSCT {
-		verifierConfig = append(verifierConfig, verify.WithSignedCertificateTimestamps(1))
-	}
-
-	if ignoreTlog && !useSignedTimestamps {
-		verifierConfig = append(verifierConfig, verify.WithCurrentTime())
-	}
-
-	// Check if artifactRef is a digest or a file path
-	var artifactOpt verify.ArtifactPolicyOption
-	if _, err := os.Stat(artifactRef); err != nil {
-		hexAlg, hexDigest, ok := strings.Cut(artifactRef, ":")
-		if !ok {
-			return nil, err
-		}
-		digestBytes, err := hex.DecodeString(hexDigest)
-		if err != nil {
-			return nil, err
-		}
-		artifactOpt = verify.WithArtifactDigest(hexAlg, digestBytes)
-	} else {
-		// Perform verification
-		payload, err := payloadBytes(artifactRef)
-		if err != nil {
-			return nil, err
-		}
-		artifactOpt = verify.WithArtifact(bytes.NewBuffer(payload))
-	}
-
-	sev, err := verify.NewSignedEntityVerifier(trustedmaterial, verifierConfig...)
-	if err != nil {
-		return nil, err
-	}
-
-	return sev.Verify(bundle, verify.NewPolicy(artifactOpt, identityPolicies...))
-}
-
 func AssembleNewBundle(ctx context.Context, sigBytes, signedTimestamp []byte, envelope *dsse.Envelope, artifactRef string, cert *x509.Certificate, ignoreTlog bool, sigVerifier signature.Verifier, pkOpts []signature.PublicKeyOption, rekorClient *client.Rekor) (*sgbundle.Bundle, error) {
```

Comment:
For a later PR, we should move this to somewhere under pkg/ rather than be under the cli package. It looks like this is only used by the `cosign bundle` command.